src/pudl/transform/eiaaeo.py

```diff
@@ -0,0 +1,324 @@
+"""Transform raw AEO tables into normalized assets.
+
+Raw AEO tables often contain many different types of data which are split out
+along different dimensions. For example, one table may contain generation
+split out by fuel type as well as prices split out by service category.
+
+As a result, we need to split these large tables into smaller tables that have
+more uniform data, which we do by filtering the large table to its relevant
+subsets, and then transforming some human-readable string fields into useful
+metadata fields.
+"""
+
+import numpy as np
+import pandas as pd
+from dagster import asset
+
+from pudl.helpers import apply_pudl_dtypes
+
+
+def __sanitize_string(series: pd.Series) -> pd.Series:
+    return series.str.lower().str.strip().str.replace(r"\W+", "_", regex=True)
```

Comment:
not required to swap or anything but there is a `pudl.helper.cleanstrings_snake` which i does this to a list of columns not just one series.

Reply:
Turns out this broke a bunch of stuff by coercing `pd.NA` to `"none"` as well as not snaking out the `/`s in the various names. So I kept `__sanitize_string` around.